common/djangoapps/util/enterprise_helpers.py

```diff
@@ -120,3 +122,50 @@ def insert_enterprise_pipeline_elements(pipeline):
 
     for index, element in enumerate(additional_elements):
         pipeline.insert(insert_point + index, element)
+
+
+def get_enterprise_customer_logo_url(request):
+    """
+    Client API operation adapter/wrapper
+    """
+    logo_url = None
+
+    if not enterprise_enabled():
+        return None
+
+    parameter = get_enterprise_branding_filter_param(request)
+    if parameter:
+        provider_id = parameter.get('provider_id', None)
+        ec_uuid = parameter.get('ec_uuid', None)
+
+        branding_info = enterprise_api.get_enterprise_branding_info(provider_id=provider_id, ec_uuid=ec_uuid)
+        if branding_info and branding_info.logo:
+            logo_url = branding_info.logo.url
+
+    return logo_url
+
+
+def set_enterprise_branding_filter_param(request, provider_id):
+    """
+    setting 'EC_BRANDING_FILTER_PARAM' in session.
+    'EC_BRANDING_FILTER_PARAM' either be provider_id or ec_uuid
+    e.g. {provider_id: 'xyz'} or {ec_src: enterprise_customer_uuid}
+    """
+
+    if EC_BRANDING_FILTER_PARAM not in request.session:
+        # SSO based Enterprise customers provide SAML idp e.g. provider_id
+        if provider_id:
```

Comment:
I am not sure if this scenario will occur or not -- it seems that it might be an exceptional situation which we are not prepared for -- in this case we are saying that `provider_id` should have priority over `ec_src`.  I am just questioning if we should either 1) raise an exception in the case where both parameters are provided, 2) log the case so we are aware, or 3) keep as-is / do nothing.